Add Files, Clipboard, and first Advanced Apps Patterns (#7645)
GoogleChrome/web.dev: src/site/content/en/patterns/clipboard/paste-text/index.md

---
layout: pattern
title: How to paste text
date: 2022-07-28
authors:

>   - charistheodoulou

Author: ```suggestion
  - harrytheodoulou
```

---
layout: pattern
title: How to paste text
date: 2022-07-28
authors:
  - harrytheodoulou
height: 800
description: >
  Pasting text into the browser consists of using the readText() method of the Async Clipboard API.
---

## The modern way

### Using the Async Clipboard API

> To read text from the user's clipboard programmatically, i.e., after a button click, you can use the [`readText()`](https://developer.mozilla.org/docs/Web/API/Clipboard/readText) method of the [Async Clipboard API](https://developer.mozilla.org/docs/Web/API/Clipboard_API). If permissions to read clipboard have not been granted yet, the call to `navigator.clipboard.readText()` will request it upon the first call to the method.

Reviewer: ```suggestion
To read text from the user's clipboard programmatically, for example, after a button click, you can use the [`readText()`](https://developer.mozilla.org/docs/Web/API/Clipboard/readText) method of the [Async Clipboard API](https://developer.mozilla.org/docs/Web/API/Clipboard_API). If permissions to read clipboard have not been granted yet, the call to `navigator.clipboard.readText()` will request it upon the first call to the method.
```
